fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalTablesProcDir.java

```diff
@@ -0,0 +1,89 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+package com.starrocks.common.proc;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.DdlException;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.MetadataMgr;
+import com.starrocks.sql.analyzer.SemanticException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExternalTablesProcDir implements ProcDirInterface {
+
+    public static final ImmutableList<String> TITLE_NAMES = new ImmutableList.Builder<String>()
+            .add("TableName").add("Type").add("Comment")
+            .build();
+
+    private String catalogName;
+    private String dbName;
+
+    public ExternalTablesProcDir(String catalogName, String dbName) {
+        this.catalogName = catalogName;
+        this.dbName = dbName;
+    }
+
+    @Override
+    public boolean register(String name, ProcNodeInterface node) {
+        return false;
+    }
+
+    @Override
+    public ProcNodeInterface lookup(String name) throws AnalysisException {
+        if (Strings.isNullOrEmpty(name)) {
+            throw new AnalysisException("name is null");
+        }
+        MetadataMgr metadataMgr = GlobalStateMgr.getCurrentState().getMetadataMgr();
+        Database db = metadataMgr.getDb(catalogName, dbName);
+        if (db == null) {
+            throw new AnalysisException("db: " + dbName + " not exists");
+        }
+        db.readLock();
+        Table tbl = null;
+        try {
+            tbl = metadataMgr.getTable(catalogName, dbName, name);
+            if (tbl == null || tbl.isOlapTable()) {
+                throw new AnalysisException("table : " + name + " not exists, or is not external table");
+            }
+        } finally {
+            db.readUnlock();
+        }
+        return new ExternalTableProcDir(tbl);
+    }
+
+    @Override
+    public ProcResult fetchResult() throws AnalysisException {
+        MetadataMgr metadataMgr = GlobalStateMgr.getCurrentState().getMetadataMgr();
+        Preconditions.checkNotNull(metadataMgr);
+        List<String> tables = null;
+        try {
+            tables = metadataMgr.listTableNames(catalogName, dbName);
+        } catch (DdlException e) {
+            throw new SemanticException(String.format("Get external tables error: ", e.getMessage()));
+        }
+        // get info
+        List<List<Comparable>> tableInfos = new ArrayList<List<Comparable>>();
+        for (String tableName : tables) {
+            Table table = metadataMgr.getTable(catalogName, dbName, tableName);
+            if (table == null) {
+                continue;
+            }
+            List<Comparable> tableInfo = new ArrayList<Comparable>();
```

Comment:
getTable will force hive metacache to load table. so if you only want to get type and comment in here, I think it's unnecessary. It's enough to list table names in here.